resources/secretsmanager_secrets.go

```diff
@@ -152,7 +152,7 @@ func fetchSecretsmanagerSecrets(ctx context.Context, meta schema.ClientMeta, _ *
 
 		// get more details about the secret
 		for _, n := range response.SecretList {
```

Comment:
would create a list of secret entries and pass them as one to channel instead of one by one, that would have also avoided this PK issue, you can keep this fix as well.